lib/mote_sms/transports/swisscom_transport.rb

```diff
@@ -0,0 +1,196 @@
+      raise ServiceError, "endpoint did respond with #{resp.code} and #{resp.body}" unless resp.code.to_i == 200
+      self.class.logger.debug resp.body
+    end
+
+    private
+
+    # Internal: Prepare request including body, headers etc.
+    #
+    # uri - The URI from the endpoint.
+    # params - The Array wifromth the attributes.
+    #
+    # Returns Net::HTTP::Post instance.
+    def http_request(from, params)
+      Net::HTTP::Post.new('/v1/messaging/sms/outbound/tel:{from}/requests').tap do |request|
+        request.body = params.to_json
+        request.content_type = 'application/json; charset=utf-8'
+        request['Accept'] = 'application/json; charset=utf-8'
+        request['client_id'] = api_key
+      end
+    end
+
+    # Internal: Build new Net::HTTP instance, enables SSL if requested.
+    #
+    # options - The Hash with all options
+    #
+    # Returns Net::HTTP client instance.
+    def http_client(options)
+      Net::HTTP.new(endpoint.host, endpoint.port).tap do |http|
+        if endpoint.instance_of?(URI::HTTPS)
+          cert = self.class.fingerprint_cert(endpoint.host)
+          http.use_ssl = true
+          http.verify_callback = ->(ok, store) { verify_fingerprint(cert.serial, ok, store) } if cert
+          options[:ssl].call(http) if options[:ssl].respond_to?(:call)
+        end
+      end
+    end
+
+    # Public: Verify SSL server certifcate when a certificate is available in
+    # mote_sms/ssl_certs/{host}.pem. Implemented to return false if first
+    # certificate in chain does not match the expected serial.
+    #
+    # serial - The expected server certificates serial
+    # ok - The Boolean forwarded by verify_callback
+    # store - The OpenSSL::X509::Store instance with the chain
+    #
+    # Returns Boolean
+    def verify_fingerprint(serial, ok, store)
+      return false unless store.chain.first.serial == serial
+      ok
+    end
+
+    # Internal: Merge defaults from class and instance with options
+    # supplied to #deliver. Removes `:http` options, because those
+    # are only for the HTTP client to set ssl verify mode et all.
+    #
+    # options - The Hash to merge with #defaults and #options.
+    #
+    # Returns Hash.
+    def prepare_options(options)
+      options = self.class.defaults.merge(self.options).merge(options)
+    end
+
+    # Internal: Prepare parameters for sending POST to endpoint, merges defaults,
+    # local and per call options, adds message related informations etc etc.
+    #
+    # message - The MoteSMS::Message to create the POST body for.
+    # options - The Hash with additional, per delivery options.
+    #
+    # Returns Array with params.
+    def post_params(message, options)
+      {
+        outboundSMSMessageRequest: {
+          senderAddress: "tel:#{message.from.to_number}",
+          address: prepare_numbers(message.to),
+          outboundSMSTextMessage: { message: message.body },
+          clientCorrelator: options[:messageid]
```

Comment:
dunno if one is required or not. Worst case, it's removed by the compact the line after